internal/controller/biossettingsset_controller.go

```diff
@@ -0,0 +1,357 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and IronCore contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package controller
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilvalidation "k8s.io/apimachinery/pkg/util/validation"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	"github.com/go-logr/logr"
+	"github.com/ironcore-dev/controller-utils/clientutils"
+	metalv1alpha1 "github.com/ironcore-dev/metal-operator/api/v1alpha1"
+)
+
+// BIOSSettingsSetReconciler reconciles a BIOSSettingsSet object
+type BIOSSettingsSetReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+const biosSettingsSetFinalizer = "metal.ironcore.dev/biosSettingsSet"
```

Comment:
In the other `Set` controllers we went with something similar to `const BIOSSettingsSetFinalizer = "metal.ironcore.dev/biossettingsset"`.